.buildkite/pipeline.yml

```diff
@@ -57,15 +66,18 @@ steps:
       queue: "{{ matrix }}"
     env:
       # See https://playwright.dev/docs/ci#debugging-browser-launches
-      DEBUG: 'pw:browser'
+      DEBUG: "pw:browser"
     matrix:
       - mac
       #- windows
+    notify:
+      - github_commit_status:
+          context: E2E Tests on {{ matrix }}
```

Comment:
Odd... `{{ matrix }}` is not resolved

<img width="520" alt="image" src="https://github.com/user-attachments/assets/85677f4f-7f5f-4ee7-9093-7ccafe1aeec3">

Reply:
Update, still odd, but at least [expected](https://buildkite.com/docs/pipelines/build-matrix):

![image](https://github.com/user-attachments/assets/36b6cc49-449d-4bb4-9a81-6e1b1663a64a)

Reply:
I experimented with different syntax here https://github.com/wordpress-mobile/WordPress-iOS/pull/23724 (I had forgotten this repo is open source, too) but didn't go anywhere.

I might write in the forums later.

In the meantime, I'll remove the status reporting for the matrix steps.

Reply:
Do you know what happens if you set a `notify.github_commit_status` to the step having the `matrix` in practice?
1. Does it only go green once the whole matrix has finished running and each of its jobs went green?
2. Or does each job that is spawn in the matrix emits the same commit status each (e.g. if the first job goes red then status is sent as red and if the next job in the matrix finishes later and ends up reporting green it will re-emit the commit status as green)?

The behavior 2 would not make much sense in practice, but might still be what happens if it's not intended behavior but a big in Buildkite.

The fact that the `{{ matrix }}` in the status name is not resolved and the un-usefulness of behavior 2 makes me think it's likely behavior 1 instead, which would probably make more sense in practice anyway.

So maybe it makes sense to have a single commit status reported for the whole matrix (and you'd just not include `{{ matrix }}` in its `context` because it would reflect the status of the whole matrix)… but even if we decide on that, I'd still want us to verify it's not behavior 2 and a bug before adopting it.

Reply:
> I might write in the forums later.

See https://forum.buildkite.community/t/support-matrix-interpolation-in-github-commit-status-context/4004